packages/repository/src/__tests__/unit/repositories/relations-helpers/flatten-targets-of-one-to-one-relation.helpers.unit.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {flattenTargetsOfOneToOneRelation} from '../../../..';
+import {
+  createCategory,
+  createProduct,
+  createManufacturer,
+} from './relations-helpers-fixtures';
+
+describe('flattenTargetsOfOneToOneRelation', () => {
+  describe('uses reduceAsSingleItem strategy for belongsTo relation', () => {
+    it('gets the result of passing in a single sourceId', () => {
+      const stationery = createCategory({id: 1, name: 'stationery'});
+      const pen = createProduct({name: 'pen', categoryId: stationery.id});
+      createProduct({name: 'eraser', categoryId: 2});
+
+      const result = flattenTargetsOfOneToOneRelation(
+        [pen.categoryId],
+        [stationery],
+        'id',
+      );
+      expect(result).to.eql([stationery]);
+    });
+
+    it('gets the result of passing in multiple sourceIds', () => {
+      const stationery = createCategory({id: 1, name: 'stationery'});
+      const book = createCategory({id: 2, name: 'book'});
+      const pen = createProduct({name: 'pen', categoryId: stationery.id});
+      const pencil = createProduct({
+        name: 'pencil',
+        categoryId: stationery.id,
+      });
+      const erasers = createProduct({name: 'eraser', categoryId: book.id});
+      // the order of sourceIds matters
+      const result = flattenTargetsOfOneToOneRelation(
+        [erasers.categoryId, pencil.categoryId, pen.categoryId],
+        [book, stationery, stationery],
```

Comment:
Hmm, a question for `flattenTargetsOfOneToOneRelation`, does it mean if I provide the second parameter in a random order, like `[stationery, book, stationery]`, it still returns an array of the results according to  the order of `[erasers.categoryId, pencil.categoryId, pen.categoryId]`?

I am very confused about what this function does, could you elaborate more about it?

Reply:
Sure. This API doc of this function: 
```
Returns an array of instances. The order of arrays is based on
the order of sourceIds
``` 
The second parameter is just a group of entities that we found by the foreign key (all related models). So this function basically just matches each the passed in sourceId with its related instance.
And you will also find a function called `flattenTargetsOfOneToManyRelation` in has many relation, it does the same thing except that one sourceId might have more than one instances.